get-terraform-provider-versions/get_terraform_provider_versions/get-tf-provider-versions.py

```diff
@@ -0,0 +1,337 @@
+                            file_path: str = os.path.join(root, filename)
+                            if file_path.find(".terragrunt-cache") == -1 and file_path.find(".git") == -1:
+                                if file_path.endswith(".tf"):
+                                    used_providers: TerraformProviders = parse_terraform_file(sub_directory_path, sub_directory, file_path, used_providers)
+                                    break
+                                else:
+                                    continue
+    else:
+        token: str = os.environ.get('GITHUB_OAUTH2_TOKEN')
+        if token is None:
+            logging.error('A GitHub OAUTH2 Token has not been defined in the GITHUB_OAUTH2_TOKEN environment variable.')
+            logging.error('The script cannot continue and will now terminate.')
+            sys.exit(3)
+        repo: Repo = Repo(token=token, owner=GITHUB_ORGANIZATION)
+
+        if (len(process_repository) == 0):
+            repo_list: list = repo.get_all_repos()
+        else:
+            repo_list: list = []
+            for github_repository in process_repository:
+                repo_exist: bool = (repo.does_repo_exist(github_repository))
+                if repo_exist is False:
+                    logging.error('The script cannot continue because the Repository specified using the -r parameter could not be found at Github.')
+                    sys.exit(4)
+                else:
+                    repo_list.append(repo.get_repo(github_repository))
+
+        for repos in repo_list:
+            for r in repos:
+                name: str = r.get('name')
+
+                if name in excluded_repos:
+                    logging.info(
+                        f'Skipping repository {name}, because it has been explicitely excluded using the -e parameter.')
+                else:
+                    is_archived: bool = r.get('archived', False)
+                    is_disabled: bool = r.get('disabled', False)
+
+                    if is_archived:
+                        logging.info(f'Skipping repository {name}, because this repo has been archived.')
+                    elif is_disabled:
+                        logging.info(f'Skipping repository {name}, because this repo has been disabled.')
+                    else:
+                        repo_has_hcl = repo.does_repo_contain_hcl(name)
+                        if repo_has_hcl:
+                            logging.info(f'Performing Terraform provider analysis on the repository {name}.')
+
+                            f = tempfile.TemporaryDirectory()
+
+                            repository_name: str = r.get('name')
+
+                            clone_url: str = r.get('clone_url')
+                            clone_command: str = f'git clone {clone_url} {f.name} -q'
+                            logging.info('\tCreating a local Clone of the GitHub repository.')
+                            subprocess.call(clone_command, shell=True)
```

Comment:
I think the recommendation in Python 3 is to use subprocess.run() instead of subprocess.call(). Reference: https://docs.python.org/3/library/subprocess.html 

It is also common to use a list of arguments instead of a string in the call/run, so 

```
clone_command: str = f'git clone {clone_url} {f.name} -q'
subprocess.call(clone_command, shell=True)
```

could be changed to 

```
clone_command: str = f'git clone {clone_url} {f.name} -q'
subprocess.call(clone_command.split(" "), shell=True)
```

Reply:
Updated and tested - works so we'll go with the recommendation from Python.